contracts/SportMarkets/Parlay/ParlayVerifier.sol

```diff
@@ -41,77 +46,424 @@ contract ParlayVerifier {
+                        (odds[i], odds[j / 2], sgpFees[i], sgpFees[j / 2]) = _getSGPSingleOdds(
+                            params.parlayPolicy.getMarketDefaultOdds(params.sportMarkets[i], params.positions[i]),
+                            params.parlayPolicy.getMarketDefaultOdds(params.sportMarkets[j / 2], params.positions[j / 2]),
+                            params.positions[i],
+                            params.positions[j / 2],
+                            sgpFee,
+                            ISportPositionalMarket(params.sportMarkets[j / 2]).optionsCount()
+                        );
+                    }
                 }
             }
-
-            (cachedTeams[lastCachedIdx].tag1, cachedTeams[lastCachedIdx].tag2) = (tag1, tag2);
-            cachedTeams[lastCachedIdx++].gameId = gameIdHome;
-            (cachedTeams[lastCachedIdx].tag1, cachedTeams[lastCachedIdx].tag2) = (tag1, tag2);
-            cachedTeams[lastCachedIdx++].gameId = gameIdAway;
+            cachedTeams[lastCachedIdx++].gameId = homeId;
+            cachedTeams[lastCachedIdx++].gameId = awayId;
+            if (odds[i] == 0) {
+                odds[i] = params.parlayPolicy.getMarketDefaultOdds(params.sportMarkets[i], params.positions[i]);
+            }
         }
     }
 
-    function _calculateRisk(
-        address[] memory _sportMarkets,
-        uint _sUSDInRisky,
-        address _parlayAMM
-    ) internal view returns (bool riskFree) {
-        // address[] memory sortedAddresses = new address[](_sportMarkets.length);
-        // sortedAddresses = _sort(_sportMarkets);
-        require(_checkRisk(_sportMarkets, _sUSDInRisky, _parlayAMM), "RiskPerComb exceeded");
-        riskFree = true;
+    /// @notice Get the Game ids and calculate the hash of each team name
+    /// @param consumer the Therundownconsumer contract
+    /// @param sportMarket The sport market to obtain the teams details
+    /// @return home the hash of the name of the home team
+    /// @return away the hash of the name of the away team
+    function _getGameIds(ITherundownConsumer consumer, address sportMarket)
+        internal
+        view
+        returns (bytes32 home, bytes32 away)
+    {
+        ITherundownConsumer.GameCreate memory game = consumer.getGameCreatedById(consumer.gameIdPerMarket(sportMarket));
+
+        home = keccak256(abi.encodePacked(game.homeTeam));
+        away = keccak256(abi.encodePacked(game.awayTeam));
+    }
+
+    /// @notice Calculate the sgpFees for the positions of two sport markets, given their odds and default sgpfee
+    /// @param odds1 the odd of position 1 (usually the moneyline odd)
+    /// @param odds2 the odd of position 2 (usually the totals/spreads odd)
+    /// @param sgpFee the default sgp fee
+    /// @return resultOdds1 the odd1
+    /// @return resultOdds2 the odd2
+    /// @return sgpFee1 the fee for position 1 or odd1
+    /// @return sgpFee2 the fee for position 2 or odd2
+    function _getSGPSingleOdds(
```

Comment:
damn this method :D this is for me too much atm
I need some kind of presentation as @danijelthales said we need to do